### PR TITLE
[RealmBrowser] Reload after external commits have been made

### LINKED ISSUE #1169: Realm Browser auto-refresh
It would be nice if the Realm Browser would auto refresh. As it is I have to close the window and reopen the Realm file each time I want to see the results of a change.

Thanks.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1169.

\c @tgoyne @jpsim

### CODE REVIEW

File: tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m [hunk 1]
```diff
@@ -55,6 +61,12 @@ - (instancetype)initWithContentsOfURL:(NSURL *)absoluteURL ofType:(NSString *)ty
             if ([realmNode connect:&error]) {
                 ws.presentedRealm  = realmNode;
 
+                ws.changeNotificationToken = [realmNode.realm addNotificationBlock:^(NSString *notification, RLMRealm *realm) {
+                    for (RLMRealmBrowserWindowController *windowController in ws.windowControllers) {
```
Comment (tgoyne) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m]: It should be possible to reload only the window for the realm that changed here.

Comment (segiddins) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m]: I believe the `RLMDocument`'s `windowControllers` property only holds the windows relevant for the given realm?

Comment (tgoyne) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m]: Ah, :+1:  then.

**Comment by jpsim** [comment]
changelog entry in enhancements, then :+1:

**Comment by segiddins** [comment]
@jpsim changelog added

File: tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m [hunk 2]
```diff
@@ -24,6 +24,12 @@
 #import "RLMRealmOutlineNode.h"
 #import "RLMRealmBrowserWindowController.h"
 
+@interface RLMDocument ()
+
+@property (nonatomic, retain) RLMNotificationToken *changeNotificationToken;
```
Comment (jpsim) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m]: `retain`? this is ARC man, hold `strong`ly.

Comment (segiddins) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMDocument.m]: done

**Pull request merged**